### ISSUE TITLE
Adding custom JS

### ISSUE THREAD

**Issue by jrGabrillo** [frappe/frappe_docker#988]
Can we add a custom JS that will run inside the admin panel?

Like, add JS that will run from the desk to any other part of the admin panel. 

I have added custom JS using Client Script that is running on list and form. This time, I want a JS to run once the Admin user has successfully logged in and redirected to the dashboard/desk.

**Comment by revant** [comment]
in context of containers use the custom_app docs or https://github.com/castlecraft/custom_frappe_docker

to know how to build js assets check frappe frameworks documentation

**Comment by jrGabrillo** [comment]
Not helpful. I tried installing the custom_frappe_docker but it doesn't come out in the admin panel. 
Can you share a good video or documentation aside from the one that you shared? 

I have successfully loaded a custom app. wiki and POS-Awesome. 
But it doesn't show in the admin panel. I also tried pushing the image created but it failed. This is the command: docker buildx bake --push

**Comment by jrGabrillo** [comment]
docker buildx bake -f docker-bake.hcl  --push
[+] Building 5.4s (17/17) FINISHED
 => [internal] load build definition from frontend.Dockerfile                                                                                                                                                                                                              0.0s
 => => transferring dockerfile: 101B                                                                                                                                                                                                                                       0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                          0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                            0.0s
 => [internal] load metadata for docker.io/frappe/bench:latest                                                                                                                                                                                                             2.3s
 => [internal] load metadata for docker.io/frappe/frappe-nginx:v14.14.3                                                                                                                                                                                                    2.3s
 => [assets 1/7] FROM docker.io/frappe/bench:latest@sha256:a786f8febcef86691c0642996a374c0fdf3051fe700fe5bafd7907b9782b882f                                                                                                                                                0.0s
 => [stage-1 1/3] FROM docker.io/frappe/frappe-nginx:v14.14.3@sha256:653e26016aa5a964e5350fb468b80e61b8798ccfefab5667f70d0aadea12531c                                                                                                                                      0.0s
 => [internal] load build context                                                                                                                                                                                                                                          0.1s
 => => transferring context: 59.22kB                                                                                                                                                                                                                                       0.0s
 => CACHED [stage-1 2/3] RUN rm -fr /usr/share/nginx/html/assets                                                                                                                                                                                                           0.0s
 => CACHED [assets 2/7] RUN bench init --version=v14.14.3 --skip-redis-config-generation --verbose --skip-assets /home/frappe/frappe-bench                                                                                                                                 0.0s
 => CACHED [assets 3/7] WORKDIR /home/frappe/frappe-bench                                                                                                                                                                                                                  0.0s
 => CACHED [assets 4/7] RUN bench get-app --branch=v14.6.0 --skip-assets --resolve-deps erpnext                                                                                                                                                                            0.0s
 => CACHED [assets 5/7] COPY --chown=frappe:frappe repos apps                                                                                                                                                                                                              0.0s
 => CACHED [assets 6/7] RUN bench setup requirements                                                                                                                                                                                                                       0.0s
 => CACHED [assets 7/7] RUN bench build --production --verbose --hard-link                                                                                                                                                                                                 0.0s
 => CACHED [stage-1 3/3] COPY --from=assets /home/frappe/frappe-bench/sites/assets /usr/share/nginx/html/assets                                                                                                                                                            0.0s
 => exporting to image                                                                                                                                                                                                                                                     0.0s
 => => exporting layers                                                                                                                                                                                                                                                    0.0s
 => => writing image sha256:afab9f8e6a3e001112a022e9a3c92d377d10974b40d5d874b7978de3e60bde5c                                                                                                                                                                               0.0s
 => => naming to docker.io/custom_images/custom_nginx:latest                                                                                                                                                                                                               0.0s
 => ERROR pushing custom_images/custom_nginx:latest with docker                                                                                                                                                                                                            2.9s
 => => pushing layer 7f86fad607ac                                                                                                                                                                                                                                          2.1s
 => => pushing layer 45ed39482f5f                                                                                                                                                                                                                                          2.1s
 => => pushing layer dfec13538195                                                                                                                                                                                                                                          2.1s
 => => pushing layer 7d1a4b0e06e0                                                                                                                                                                                                                                          2.1s
 => => pushing layer 94962943e7f4                                                                                                                                                                                                                                          2.1s
 => => pushing layer d02b7e08b07e                                                                                                                                                                                                                                          2.3s
 => => pushing layer f7a8216b0977                                                                                                                                                                                                                                          2.3s
 => => pushing layer 2ba9c5930a3b                                                                                                                                                                                                                                          2.3s
 => => pushing layer 362559305ba5                                                                                                                                                                                                                                          2.3s
 => => pushing layer 435befd668f2                                                                                                                                                                                                                                          2.3s
 => => pushing layer 7c5d06846c68                                                                                                                                                                                                                                          2.3s
 => => pushing layer 3ac68311a153                                                                                                                                                                                                                                          2.3s
 => => pushing layer 994393dc58e7                                                                                                                                                                                                                                          2.3s
------
 > pushing custom_images/custom_nginx:latest with docker:
------
ERROR: denied: requested access to the resource is denied

**Comment by jrGabrillo** [comment]
```
docker buildx bake -f docker-bake.hcl  --push
[+] Building 5.4s (17/17) FINISHED
 => [internal] load build definition from frontend.Dockerfile                                                                                                                                                                                                              0.0s
 => => transferring dockerfile: 101B                                                                                                                                                                                                                                       0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                          0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                            0.0s
 => [internal] load metadata for docker.io/frappe/bench:latest                                                                                                                                                                                                             2.3s
 => [internal] load metadata for docker.io/frappe/frappe-nginx:v14.14.3                                                                                                                                                                                                    2.3s
 => [assets 1/7] FROM docker.io/frappe/bench:latest@sha256:a786f8febcef86691c0642996a374c0fdf3051fe700fe5bafd7907b9782b882f                                                                                                                                                0.0s
 => [stage-1 1/3] FROM docker.io/frappe/frappe-nginx:v14.14.3@sha256:653e26016aa5a964e5350fb468b80e61b8798ccfefab5667f70d0aadea12531c                                                                                                                                      0.0s
 => [internal] load build context                                                                                                                                                                                                                                          0.1s
 => => transferring context: 59.22kB                                                                                                                                                                                                                                       0.0s
 => CACHED [stage-1 2/3] RUN rm -fr /usr/share/nginx/html/assets                                                                                                                                                                                                           0.0s
 => CACHED [assets 2/7] RUN bench init --version=v14.14.3 --skip-redis-config-generation --verbose --skip-assets /home/frappe/frappe-bench                                                                                                                                 0.0s
 => CACHED [assets 3/7] WORKDIR /home/frappe/frappe-bench                                                                                                                                                                                                                  0.0s
 => CACHED [assets 4/7] RUN bench get-app --branch=v14.6.0 --skip-assets --resolve-deps erpnext                                                                                                                                                                            0.0s
 => CACHED [assets 5/7] COPY --chown=frappe:frappe repos apps                                                                                                                                                                                                              0.0s
 => CACHED [assets 6/7] RUN bench setup requirements                                                                                                                                                                                                                       0.0s
 => CACHED [assets 7/7] RUN bench build --production --verbose --hard-link                                                                                                                                                                                                 0.0s
 => CACHED [stage-1 3/3] COPY --from=assets /home/frappe/frappe-bench/sites/assets /usr/share/nginx/html/assets                                                                                                                                                            0.0s
 => exporting to image                                                                                                                                                                                                                                                     0.0s
 => => exporting layers                                                                                                                                                                                                                                                    0.0s
 => => writing image sha256:afab9f8e6a3e001112a022e9a3c92d377d10974b40d5d874b7978de3e60bde5c                                                                                                                                                                               0.0s
 => => naming to docker.io/custom_images/custom_nginx:latest                                                                                                                                                                                                               0.0s
 => ERROR pushing custom_images/custom_nginx:latest with docker                                                                                                                                                                                                            2.9s
 => => pushing layer 7f86fad607ac                                                                                                                                                                                                                                          2.1s
 => => pushing layer 45ed39482f5f                                                                                                                                                                                                                                          2.1s
 => => pushing layer dfec13538195                                                                                                                                                                                                                                          2.1s
 => => pushing layer 7d1a4b0e06e0                                                                                                                                                                                                                                          2.1s
 => => pushing layer 94962943e7f4                                                                                                                                                                                                                                          2.1s
 => => pushing layer d02b7e08b07e                                                                                                                                                                                                                                          2.3s
 => => pushing layer f7a8216b0977                                                                                                                                                                                                                                          2.3s
 => => pushing layer 2ba9c5930a3b                                                                                                                                                                                                                                          2.3s
 => => pushing layer 362559305ba5                                                                                                                                                                                                                                          2.3s
 => => pushing layer 435befd668f2                                                                                                                                                                                                                                          2.3s
 => => pushing layer 7c5d06846c68                                                                                                                                                                                                                                          2.3s
 => => pushing layer 3ac68311a153                                                                                                                                                                                                                                          2.3s
 => => pushing layer 994393dc58e7                                                                                                                                                                                                                                          2.3s
------
 > pushing custom_images/custom_nginx:latest with docker:
------
ERROR: denied: requested access to the resource is denied
```

**Comment by revant** [comment]
> > pushing custom_images/custom_nginx:latest with docker

this location will get access denied

name the image and registery in such way that it can be pushed

**Comment by jrGabrillo** [comment]
I still cant push the custom app. This time, I am trying to push POS-awesome and wiki to see if it will really be pushed in the erpnext. 

![Screenshot 2022-11-11 at 5 38 57 PM](https://user-images.githubusercontent.com/19164950/201312750-a0742e1f-775f-403b-bfd0-5120e6b38d8c.png)

**Comment by jrGabrillo** [comment]
Do i really need to do this if what I just want to do is add a js that will run on admin panel only?

**Comment by jrGabrillo** [comment]
> to know how to build js assets check frappe frameworks documentation

Can you share a link to follow? I am confused rn. 

Thanks btw for helping me.

**Comment by jrGabrillo** [comment]
<img width="785" alt="Screenshot 2022-11-11 at 11 01 37 PM" src="https://user-images.githubusercontent.com/19164950/201367131-c0cbb9b6-e7d7-41c4-b42b-3c0986286432.png">
Which file I should edit to add this code snippet so I can run a JS code inside admin panel?

**Comment by jrGabrillo** [comment]
Hi @revant 
I successfully integrated a JS file in the ERPNext. 

Here's what I did. 
1. I deployed my js code in github then created a cdn link using https://raw.githack.com. So, basically, I just pasted my publicly available js code in githack to create a cdn link. 
2. Then I access the frappe's app.html located in /home/frappe/frappe-bench/apps/frappe/frappe/www using docker compose. 
3. Inserted a script tag before the body's end tag. 

Let me know if this workaround has ok or not. Right now, everything works just as what I expected.
